frontend/src/container/GridGraphLayout/Graph/FullView/types.ts

```diff
@@ -0,0 +1,75 @@
+import { CheckboxChangeEvent } from 'antd/es/checkbox';
+import { ChartData, ChartDataset } from 'chart.js';
+import { GraphOnClickHandler } from 'components/Graph/types';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { Widgets } from 'types/api/dashboard/getAll';
+
+export interface DataSetProps {
+	index: number;
+	data: number | null;
+	label: string;
+	borderWidth: number;
+	spanGaps: boolean;
+	animations: boolean;
+	borderColor: string;
+	showLine: boolean;
+	pointRadius: number;
+}
+
+export interface LegendEntryProps {
+	label: string;
+	show: boolean;
+}
+
+export type ExtendedChartDataset = ChartDataset & {
+	show: boolean;
+	sum: number;
+	avg: number;
+	min: number;
+	max: number;
+};
+
+export type PanelTypeAndGraphManagerVisibilityProps = {
+	[key in keyof typeof PANEL_TYPES]: boolean;
+};
+
+export interface LabelProps {
+	labelClickedHandler: (labelIndex: number) => void;
+	labelIndex: number;
+	label: string;
+}
+
+export interface GraphManagerProps {
+	data: ChartData;
+	name: string;
+	onToggleModelHandler?: () => void;
+}
+
+export interface CheckBoxProps {
+	data: ChartData;
+	index: number;
+	graphVisibilityState: boolean[];
+	checkBoxOnChangeHandler: (e: CheckboxChangeEvent, index: number) => void;
+}
+
+export interface FullViewProps {
+	widget: Widgets;
+	fullViewOptions?: boolean;
+	onClickHandler?: GraphOnClickHandler;
+	name: string;
+	yAxisUnit?: string;
+	onDragSelect?: (start: number, end: number) => void;
+	graphsVisibilityStates?: boolean[];
+	isDependedDataLoaded?: boolean;
+	onToggleModelHandler?: () => void;
```

Comment:
duplicate `onToggleModelHandler` see other keys also

Reply:
Are you suggesting making a separate type for `onToogleModelHandler` and the extend that type wherever it is useful?